onnxruntime/core/providers/cpu/math/top_k.cc

```diff
@@ -167,7 +176,7 @@ Status TopK<10, float>::Compute(OpKernelContext* p_op_kernel_context) const {
   const vector<int64_t>& y_shape = Y->Shape().GetDims();
   if (y_shape.size() != 1 || y_shape[0] != 1) return Status(common::ONNXRUNTIME, common::FAIL, "k tensor should be a 1D tensor of size 1");
   unsigned parsed_input_k = gsl::narrow_cast<unsigned>(Y->template Data<int64_t>()[0]);
-  if (parsed_input_k <= 0) return Status(common::ONNXRUNTIME, common::FAIL, "value of k should be greater than 0");
+  if (parsed_input_k < 0) return Status(common::ONNXRUNTIME, common::FAIL, "value of k must not be negative");
```

Comment:
parsed_input_k is unsigned, maybe this check should happen before the cast?

Reply:
Exactly. Thanks!